signer/src/network/libp2p/network.rs

```diff
@@ -104,11 +105,26 @@ impl MessageTransfer for P2PNetwork {
     }
 
     fn receiver_stream(&self) -> BroadcastStream<SignerSignal> {
-        let (sender, receiver) = tokio::sync::broadcast::channel(10);
+        // This is the same capacity of `signal_tx` in the SignerContext,
+        // which is typically used to create this P2PNetwork.
+        let (sender, receiver) = tokio::sync::broadcast::channel(SIGNER_CHANNEL_CAPACITY);
```

Comment:
I think that if you changed this to an unbounded mpsc stream that it would work. I think the problem is that the poller of the `BroadcastStream` returned below is still polling items one-at-a-time with (sometimes long) processing times in-between each item, which allows this channel to quickly fill up especially when coordinated rounds are flying around.

But that still leaves the potential problem of OOM if this channel grows too much since it accumulates _all_ P2P messages and not only the ones the consumer is interested in, which is why I'm not sure if trying to aggregate the "inbox" up to this abstraction vs. the actual consumer is the best solution since the consumer, knowing exactly what they're interested in, can more effectively manage the items that flow into their inbox vs. get dropped and also potentially take action if they see that the inbox is getting too big.

I hope I'm making sense? :sweat_smile:

Reply:
Yeah you're making sense. 🤔